aws/aws-sdk-js-v3
# chore(codegen): update smithy-typescript commit

### Issue
Issue number, if available, prefixed with "#"

N/A.

### Description
What does this implement/fix? Explain your changes.

Bump smithy-typescript commit to https://github.com/awslabs/smithy-typescript/commit/8e23b1b1df60d81ae58a93998228a8a60a72cc93.

Dependent on:

- https://github.com/awslabs/smithy-typescript/pull/920
- https://github.com/awslabs/smithy-typescript/pull/921
- NPM package release
- `TypeScriptDependency` codegen dependencies bump PR: 8e23b1b1df60d81ae58a93998228a8a60a72cc93

### Testing
How was this change tested?

1. Change `config.js` to point to `8e23b1b1df60d81ae58a93998228a8a60a72cc93`
2. Run `yarn generate-clients`
3. Run `node scripts/runtime-dependency-version-check/runtime-dep-version-check.js`
4. Run `yarn`
5. Run `yarn build:all`
6. PR CI passes

Observed codegen changes: `HttpHandlerExtensionConfiguration` is now added to each client in:

- `extensionConfiguration.ts`

```diff
diff --git a/clients/client-accessanalyzer/src/extensionConfiguration.ts b/clients/client-accessanalyzer/src/extensionConfiguration.ts
index 6d8fbc43442..660d1c6fbfe 100644
--- a/clients/client-accessanalyzer/src/extensionConfiguration.ts
+++ b/clients/client-accessanalyzer/src/extensionConfiguration.ts
@@ -1,7 +1,10 @@
 // smithy-typescript generated code
+import { HttpHandlerExtensionConfiguration } from "@smithy/protocol-http";
 import { DefaultExtensionConfiguration } from "@smithy/types";
 
 /**
  * @internal
  */
-export interface AccessAnalyzerExtensionConfiguration extends DefaultExtensionConfiguration {}
+export interface AccessAnalyzerExtensionConfiguration
+  extends HttpHandlerExtensionConfiguration,
+    DefaultExtensionConfiguration {}
```

- `runtimeExtensions.ts`

```diff
diff --git a/clients/client-accessanalyzer/src/runtimeExtensions.ts b/clients/client-accessanalyzer/src/runtimeExtensions.ts
index 52162c3f77b..df73513b2eb 100644
--- a/clients/client-accessanalyzer/src/runtimeExtensions.ts
+++ b/clients/client-accessanalyzer/src/runtimeExtensions.ts
@@ -1,4 +1,5 @@
 // smithy-typescript generated code
+import { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig } from "@smithy/protocol-http";
 import { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig } from "@smithy/smithy-client";
 
 import { AccessAnalyzerExtensionConfiguration } from "./extensionConfiguration";
@@ -7,7 +8,7 @@ import { AccessAnalyzerExtensionConfiguration } from "./extensionConfiguration";
  * @public
  */
 export interface RuntimeExtension {
-  configure(clientConfiguration: AccessAnalyzerExtensionConfiguration): void;
+  configure(extensionConfiguration: AccessAnalyzerExtensionConfiguration): void;
 }
 
 /**
@@ -25,6 +26,7 @@ const asPartial = <T extends Partial<AccessAnalyzerExtensionConfiguration>>(t: T
 export const resolveRuntimeExtensions = (runtimeConfig: any, extensions: RuntimeExtension[]) => {
   const extensionConfiguration: AccessAnalyzerExtensionConfiguration = {
     ...asPartial(getDefaultExtensionConfiguration(runtimeConfig)),
+    ...asPartial(getHttpHandlerExtensionConfiguration(runtimeConfig)),
   };
 
   extensions.forEach((extension) => extension.configure(extensionConfiguration));
@@ -32,5 +34,6 @@ export const resolveRuntimeExtensions = (runtimeConfig: any, extensions: Runtime
   return {
     ...runtimeConfig,
     ...resolveDefaultRuntimeConfig(extensionConfiguration),
+    ...resolveHttpHandlerRuntimeConfig(extensionConfiguration),
   };
 };
```

### Additional context
Add any other context about the PR here.

N/A.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
